consensus/src/txn_notifier.rs

```diff
@@ -57,7 +57,13 @@ impl TxnNotifier for MempoolNotifier {
             return Ok(());
         }
         let compute_status = compute_results.compute_status();
-        if txns.len() + 2 != compute_status.len() {
+        let correct_len = if compute_results.has_reconfiguration() {
+            // reconfiguration blocks don't have state checkpoint
```

Comment:
that depends on if block_gas_limit is set?

Reply:
Yes. Resolved by new fix.